lib/perl5/App/Cilly.pm.in

```diff
@@ -1517,12 +1526,34 @@ sub new {
 	    # skipping -###, --help, --target-help, --version
 
 	    # C Language Options
-            "-ansi" => { TYPE => 'ALLARGS' },
-            '-std=' => { TYPE => 'ALLARGS' },
+      "-ansi" => {
+            TYPE => 'ALLARGS',
+            RUN => sub {
+                push @{$stub->{CILARGS}}, "--cstd", "c90";
+            }},
+      "-std=(c90|c89|iso9899:1990|iso9899:199409|gnu90|gnu89)" => {
+            TYPE => 'ALLARGS',
+            RUN => sub {
+                push @{$stub->{CILARGS}}, "--cstd", "c90";
+            }},
+      "-std=(c99|c9x|iso9899:1999|iso9899:199x|gnu99|gnu9x)" => {
+            TYPE => 'ALLARGS',
+            RUN => sub {
+                push @{$stub->{CILARGS}}, "--cstd", "c99";
+            }},
+      "-std=(c1x|iso9899:2011|c17|c18|iso9899:2017|iso9899:2018|c2x|gnu11|gnu1x|gnu17|gnu18|gnu2x)" => {
+            TYPE => 'ALLARGS',
+            RUN => sub {
+                push @{$stub->{CILARGS}}, "--cstd", "c11";
+            }},
+      "-fgnu89-inline" => {
+            RUN => sub {
+                push @{$stub->{CILARGS}}, "--cstd", "gnu89inline";
+            }},
```

Comment:
What happens if both `-fgnu89inline` and `-std=c99` are present. Does it simply take the later one? And is that consistent with GCC.

Reply:
If the inlining behavior is othogonal to the standard, as indicated by the GCC feature flag as well, then why do we try to cram it into a set of standards instead of keeping a separate flag.

Reply:
Indeed it was not clear, what standard is used when `gnu89inline` is stored as part of the `cstd`. I now have a separate flag for `gnu89inline` which results in using the gnu89 inline semantics no matter which standard is used. I still kept the `cstd` option, because with the older standards the gnu89 inline semantics will also be used without `-fgnu89inline`. Whether one needs the C11 option is arguable, I just thought it is easier extensible and can be used for other aspects (like the parsing of decimal constants).